### PR TITLE
Refactor reverse interface

### PR DESCRIPTION
**Pull request by fmauch**
This moves the ReverseInteface and the ScriptSender from the `comm` namespace to the `control` namespace. As we plan to extend the control protocol structure with the changes from the beta-testing branch, I think it makes sense to clean up things a little beforehand.

Note: I decided to keep the `ControlMode` enums inside the `comm` namespace first because of backwards compatibility, as they are used on application level, and second because they actually define the communication protocol and therefore seem to be not too wrong inside the comm namespace.

### CODE REVIEW

File: src/control/reverse_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+void ReverseInterface::disconnectionCallback(const int filedescriptor)
+{
+  URCL_LOG_INFO("Connection to reverse interface dropped.", filedescriptor);
+  client_fd_ = -1;
+  handle_program_state_(false);
+}
+
+void ReverseInterface::messageCallback(const int filedescriptor, char* buffer)
+{
+  URCL_LOG_WARN("Messge on ReverseInterface received. The reverse interface currently does not support any message "
```
Comment (stefanscherzinger) [inline src/control/reverse_interface.cpp:105]:
```suggestion
  URCL_LOG_WARN("Message on ReverseInterface received. The reverse interface currently does not support any message "
```

File: src/control/script_sender.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  server_.setMessageCallback(
+      std::bind(&ScriptSender::messageCallback, this, std::placeholders::_1, std::placeholders::_2));
+  server_.setConnectCallback(std::bind(&ScriptSender::connectionCallback, this, std::placeholders::_1));
+  server_.setDisconnectCallback(std::bind(&ScriptSender::disconnectionCallback, this, std::placeholders::_1));
+  server_.start();
+}
+
+void ScriptSender::connectionCallback(const int filedescriptor)
+{
+  URCL_LOG_DEBUG("New client connected at FD %d.", filedescriptor);
```
Comment (stefanscherzinger) [inline src/control/script_sender.cpp:46]:
```suggestion
  URCL_LOG_DEBUG("New client connected at file descriptor %d.", filedescriptor);
```
Is _FD_ sufficiently known as a term? If yes, then it's ok.

File: src/control/script_sender.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+void ScriptSender::disconnectionCallback(const int filedescriptor)
+{
+  URCL_LOG_DEBUG("Client at FD %d disconnected.", filedescriptor);
```
Comment (stefanscherzinger) [inline src/control/script_sender.cpp:51]:
```suggestion
  URCL_LOG_DEBUG("Client at file descriptor %d disconnected.", filedescriptor);
```

**Review by stefanscherzinger — APPROVED** [review]
I read over this and it looks fine.

File: src/control/script_sender.cpp (hunk 2 above)
Comment (fmauch) [reply src/control/script_sender.cpp:46]: It's a debug output anyway...

**Pull request merged**